### PR TITLE
Fix issue with join condition referencing a previous relation

### PR DESCRIPTION
**Pull request by matriv**
Previously, the additional dependency was not recognized as the code was only checking `ScopedSymbol`s for the case relations were aliased. This led to the join order possible changed, and when the `JoinPair`, for which the condition was containing additional relations, could become 1st in the ordering, thus the required symbols from the additional relation were not available, e.g.:
```
SELECT * FROM t1
CROSS JOIN t2
INNER JOIN t3 ON t3.x = t1.x AND t3.y = t2
```
The `JoinPair` t2, t3 will become the 1st in order but it's referencing t1 in the join condition, which is not available, as it would be joined next in order.

Fixes: #13942

### CODE REVIEW

File: server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java [hunk 1]
```diff
@@ -160,12 +161,25 @@ private static boolean hasAdditionalDependencies(JoinPair joinPair) {
...
+
+        // Aliased tables
+        if (hasAdditionalDependencies[0] == false) {
+            FieldsVisitor.visitFields(condition, scopedSymbol -> {
+                RelationName relationName = scopedSymbol.relation();
+                if (!relationName.equals(joinPair.left()) && !relationName.equals(joinPair.right())) {
+                    hasAdditionalDependencies[0] = true;
+                }
+            });
+        }
```
Comment (seut) [inline server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java:181]: Looks like both uses the same consumer, maybe the consumer could be stored and re-used?

**Review by seut — APPROVED** [review]
Nice! Added just a more minor suggestion

File: server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java (hunk 1 above)
Comment (matriv) [reply server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java:181]: It's not the same consumer, one is Reference the other ScopedSymbol, so I could extract the if part only, and I think we don't gain much for just 3 lines.

**Comment by matriv** [comment]
@Mergifyio backport 5.2

**Pull request merged**